department-of-veterans-affairs/caseflow
# APPEALS-61670: Update job to call 'distribute non genpop priority appeals' every time

Resolves [APPEALS-61670](https://jira.devops.va.gov/browse/APPEALS-61670)

# Description
Updates the `PushPriorityAppealsToJudgesJob` to specifically distribute non-genpop priority appeals **before** beginning the process of calculating and distributing the genpop priority appeals.
- Modifies the `PushPriorityAppealsToJudgesJob` to **not** skip the `distribute_non_genpop_priority_appeals` method call
  - Adds a feature toggle `:acd_ppj_distribute_not_genpop` to allow us to disable this if there is a problem in the future
- Fixes a bug in the CaseDocket common filter logic where if a legacy appeal with no hearing was held had a deciding judge on the Ineligible Judges list, it would distribute when genpop was set to `not_genpop` for that distribution, even though it is a genpop appeal
  - Adds two appeals to the case_docket and aoj_case_docket test files for the above scenario 
- Fixes the `batch_size` value for previous distributions in the seed data
- Updates and adds tests to push_priority_appeals_to_judges_job_spec.rb for distributing non-genpop appeals with the new feature toggle enabled

## Acceptance Criteria
- [x] Distribution Rspec tests are passing
- [x] PushPriorityAppealsToJudgesJob distributes all non-genpop ready priority appeals to eligible judges before calculating or distributing any genpop priority appeals

## Testing Plan
1. `make reset` or re-seed a demo environment with the "reseed DB" button
2. Run the PushPriorityAppealsToJudgesJob, either via the scheduled jobs page in the UI or via the rails console
3. Verify that appeals were distributed to their tied/affinity judge, if applicable
4. Verify that the number of distributed appeals is relatively even
5. Verify that if a judge received tied/affinity appeals, that they received fewer genpop appeals than a judge who did not receive any tied/affinity appeals
